File: bots/lab-examples/java/Starter.java
import dev.robocode.tankroyale.botapi.*;
import dev.robocode.tankroyale.botapi.events.*;

/** Starter — scan, move, fire. */
public class Starter extends Bot {
    public static void main(String[] args) { new Starter().start(); }
    @Override public void run() {
        while (isRunning()) { turnRadarRight(360); forward(50); turnLeft(20); }
    }
    @Override public void onScannedBot(ScannedBotEvent e) { fire(1.5); }
    @Override public void onHitWall(HitWallEvent e) { back(60); turnRight(90); }
}

File: bots/lab-examples/ts/Starter.ts
import { Bot, HitWallEvent, ScannedBotEvent } from "@robocode.dev/tank-royale-bot-api";

/** Starter — scan, move, fire. Baseline pra editar. */
class Starter extends Bot {
  static main() {
    new Starter().start();
  }
  override run() {
    while (this.isRunning()) {
      this.turnRadarRight(360);
      this.forward(50);
      this.turnLeft(20);
    }
  }
  override onScannedBot(_e: ScannedBotEvent) {
    this.fire(1.5);
  }
  override onHitWall(_e: HitWallEvent) {
    this.back(60);
    this.turnRight(90);
  }
}
Starter.main();
